regression/Makefile

```diff
@@ -28,6 +28,7 @@ DIRS = cbmc \
        systemc \
        contracts \
        goto-cc-file-local \
+       goto-cc-regression \
```

Comment:
:pick: Everything here is “regression” tests, so the "regression" here doesn’t give us any information. Might be better to give this a more descriptive name – either what kind of testing it is doing if you think we’re likely to add different tests with the same test script in the future, or otherwise maybe just include the github issue ID as part of the name (e.g. `-gh-issue-XXXX`).

(Well technically most of them are integration tests, but we’re calling them regression tests regardless).

Reply:
Hm, you are right, I'm not perfectly happy with the `-regression` suffix, but my original one was `-crash` and I didn't think that was much better. 

TBH having it very issue specific is something that could easily be done, but for a strange kind of fashion I don't like it.

Let me ponder on that.

Reply:
Also might be worth mentioning that originally I wanted the test to go into `goto-cc-file-local` but there was just no way I could bend the `chain.sh` to do exactly what I wanted it to do without causing issues.

Reply:
+1 what @hannes-steffenhagen-diffblue said.

( I think they are a mix of integration tests and regressions tests really but let's not get into that can of worms right now. )